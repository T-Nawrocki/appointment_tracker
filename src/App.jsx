import React, { useState } from 'react';
import AppointmentsList from './components/AppointmentsList';
import NewAppointmentForm from './components/NewAppointmentForm';
import './styles/App.css';

function App() {

  const [appointments, setAppointments] = useState([]);

  const onNewAppointmentSubmit = newAppointment => {
    setAppointments([...appointments, newAppointment]);
  };

  // TODO: refector to use ID once those are implemented
  const onDeleteAppointment = deletedAppointment => {
    setAppointments(appointments.filter(appointment => {
      return JSON.stringify(appointment) !== JSON.stringify(deletedAppointment);
    }));
  };

  return (
    <div className="App">
      <h1 id="app-heading">Appointments Tracker</h1>
      <NewAppointmentForm onNewAppointmentSubmit={onNewAppointmentSubmit} />
      <AppointmentsList appointments={appointments} onDeleteAppointment={onDeleteAppointment} />
    </div>
  );
}

export default App;
