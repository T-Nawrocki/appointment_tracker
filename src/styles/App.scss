@use 'common/colours';

.App {
  margin: 5% 0 25% 0;
  text-align: center;
}

ul { 
  padding: 0;
  list-style: none;
}

