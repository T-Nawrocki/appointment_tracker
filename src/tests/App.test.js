import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import App from '../App';

configure({ adapter: new Adapter() });

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should render the app', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the new appointment form', () => {
    expect(wrapper.find('NewAppointmentForm').length).toEqual(1);
  });

  it('should render the appointment list', () => {
    expect(wrapper.find('AppointmentsList').length).toEqual(1);
  });

  it('should be able to add an appointment', () => {
    
  });
});
