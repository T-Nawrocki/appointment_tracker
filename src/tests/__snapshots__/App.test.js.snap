// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the app 1`] = `
<div
  className="App"
>
  <h1
    id="app-heading"
  >
    Appointments Tracker
  </h1>
  <ul
    id="input-warnings"
  />
  <form
    id="new-appointment-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="new-appointment-title"
    >
      Title
    </label>
    <input
      id="new-appointment-title"
      name="new-appointment-title"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="new-appointment-date"
    >
      Date
    </label>
    <input
      id="new-appointment-date"
      min="2020-11-13"
      name="new-appointment-date"
      onChange={[Function]}
      type="date"
      value=""
    />
    <label
      htmlFor="new-appointment-time"
    >
      Time
    </label>
    <select
      id="new-appointment-time"
      name="new-appointment-time"
      onChange={[Function]}
      value=""
    >
      <option
        disabled={true}
        value=""
      >
        --:--
      </option>
      <option
        value="8:00"
      >
        8:00
      </option>
      <option
        value="8:30"
      >
        8:30
      </option>
      <option
        value="9:00"
      >
        9:00
      </option>
      <option
        value="9:30"
      >
        9:30
      </option>
      <option
        value="10:00"
      >
        10:00
      </option>
      <option
        value="10:30"
      >
        10:30
      </option>
      <option
        value="11:00"
      >
        11:00
      </option>
      <option
        value="11:30"
      >
        11:30
      </option>
      <option
        value="12:00"
      >
        12:00
      </option>
      <option
        value="12:30"
      >
        12:30
      </option>
      <option
        value="13:00"
      >
        13:00
      </option>
      <option
        value="13:30"
      >
        13:30
      </option>
      <option
        value="14:00"
      >
        14:00
      </option>
      <option
        value="14:30"
      >
        14:30
      </option>
      <option
        value="15:00"
      >
        15:00
      </option>
      <option
        value="15:30"
      >
        15:30
      </option>
      <option
        value="16:00"
      >
        16:00
      </option>
      <option
        value="16:30"
      >
        16:30
      </option>
      <option
        value="17:00"
      >
        17:00
      </option>
      <option
        value="17:30"
      >
        17:30
      </option>
    </select>
    <input
      name="new-appointment-submit"
      type="submit"
      value="Book appointment"
    />
  </form>
  <div
    id="appointments-list"
  >
    <h2>
      Appointments
    </h2>
    <div
      id="appointments-upcoming"
    >
      <h3>
        Upcoming Appointments
      </h3>
      <ul />
    </div>
    <div
      id="appointments-past"
    >
      <h3>
        Past Appointments
      </h3>
      <ul />
    </div>
  </div>
</div>
`;
